lib/compiler.js

```diff
@@ -644,19 +657,28 @@ function getExcludedExtensions() {
   };
 }
 
-function isPreferredExtension(filename) {
-  var includeExtensions = require.extensions;
+function arrayToObject(array) {
+  return array.reduce(function(obj, val) {
+    obj[val] = val;
+    return obj;
+  }, {});
+}
+
+function isPreferredExtension(filename, extensions) {
+  var includeExtensions = extensions ?
+    arrayToObject(extensions) : require.extensions;
```

Comment:
Since `isPreferredExtension` is a private method, there is no need to keep support for the old one-arg API.

```js
function isPreferredExtension(filename, includeExtensions) {
  assert(!!includeExtensions, '"includeExtensions" argument is required');
  // the code below remains unchanged
  var ext = path.extname(filename);
  return (ext in includeExtensions) && !(ext in getExcludedExtensions());}
```